### PR TITLE
Revise saasform/saasform Dockerfile

### PR DESCRIPTION
**Pull request by psmiraglia**
# Describe the scope of the PR

* Update the base image in order to use the current Debian "stable" (`buster`) and Node "active LTS" (`14.x`)
* Add `.dockerignore` file to exclude useless sources
* Revise the layers in order to make the resulting image as deterministic as possible
* Revise the compose file in order to build the images at runtime (if `--build` is provided)

_Solved_

- None

_Added_

- None

## Add any shortcut taken

Some tweaks on `data/config/saasform.yml`, but purely related to my devenv.

## Tests

- [X] I manually tested
- [ ] I added unit test
- [ ] I added e2e test
- [ ] I ran the existing unit test and they do not fail
- [ ] I ran the existing e2e test and they do not fail

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Launch Saasform:
 ```bash
 git clone https://github.com/saasform/saasform
 cd saasform
-docker-compose up
+docker-compose up --build
```
Comment (0x0ece) [inline README.md:21]: See next comment

File: docker-compose.yml [hunk 2]
```diff
@@ -14,7 +14,8 @@ services:
       - ./data/db:/var/lib/mysql:cached
 
   saasform:
-    image: saasform/saasform:latest
+    image: saasform/saasform:localbuild
```
Comment (0x0ece) [inline docker-compose.yml:17]: Let's keep this file unchanged. We prefer people to pull from Docker Hub for now, as `main` may not be stable at points. All the images we build are based on releases and tested.

File: docker-compose.yml [hunk 3]
```diff
@@ -37,7 +38,8 @@ services:
     command: ["bash", "/app/bin/startup.sh"]
 
   demo-express:
-    image: saasform/demo-express:latest
+    image: saasform/demo-express:localbuild
```
Comment (0x0ece) [inline docker-compose.yml:41]: Same as above

**Review by 0x0ece — APPROVED** [review]
Approving to speed up, but please revert docker-compose.yml and README.md

File: README.md (hunk 1 above)
Comment (davidevernizzi) [reply README.md:21]: I'm approving the PR and will do these update myself, to speed up

File: docker-compose.yml (hunk 2 above)
Comment (davidevernizzi) [reply docker-compose.yml:17]: I'm approving the PR and will do these update myself, to speed up

File: docker-compose.yml (hunk 3 above)
Comment (davidevernizzi) [reply docker-compose.yml:41]: I'm approving the PR and will do these update myself, to speed up

**Pull request merged**